src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/GTKMediaPlayer.events.cs

```diff
@@ -0,0 +1,332 @@
+using System.Timers;
+using Timer = System.Timers.Timer;
+using System.Globalization;
+using Windows.Media.Playback;
+using System.Threading.Tasks;
+using Uno.UI.Runtime.Skia;
+
+namespace Uno.UI.Media;
+
+public partial class GtkMediaPlayer
+{
+	private TaskCompletionSource<string> _taskCompletionSource = new TaskCompletionSource<string>("");
+	public event EventHandler<object>? OnSourceFailed;
+	public event EventHandler<object>? OnSourceEnded;
+	public event EventHandler<object>? OnMetadataLoaded;
+	public event EventHandler<object>? OnTimeUpdate;
+	public event EventHandler<object>? OnSourceLoaded;
+
+	private async Task Initialized()
+	{
+		if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+		{
+			this.Log().Debug($"GtkMediaPlayer Initialized");
+			this.Log().Debug($"Creating libvlc");
+		}
+		_libvlc = new LibVLC(enableDebugLogs: false);
+		if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+		{
+			this.Log().Debug($"Creating player");
+		}
+		_mediaPlayer = new LibVLCSharp.Shared.MediaPlayer(_libvlc);
+		if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+		{
+			this.Log().Debug($"Creating VideoView");
+		}
+		_videoView = new LibVLCSharp.GTK.VideoView();
+		_videoContainer = new ContentControl
+		{
+			HorizontalAlignment = HorizontalAlignment.Center,
+			VerticalAlignment = VerticalAlignment.Center,
+			HorizontalContentAlignment = HorizontalAlignment.Stretch,
+			VerticalContentAlignment = VerticalAlignment.Stretch,
+		};
+		_taskCompletionSource = new TaskCompletionSource<string>("");
+		_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+				{
+					InitializedMedia();
+				});
+		await _taskCompletionSource.Task;
+	}
+
+	private Task InitializedMedia()
+	{
+		try
+		{
+			if (_videoView != null && _mediaPlayer != null && _videoContainer != null)
+			{
+				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				{
+					this.Log().Debug($"Set MediaPlayer");
+				}
+				_videoView.MediaPlayer = _mediaPlayer;
+
+				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				{
+					this.Log().Debug($"Content _videoView on Dispatcher");
+				}
+				_videoView.Visible = true;
+				_videoView.MediaPlayer = _mediaPlayer;
+
+				_mediaPlayer.TimeChanged += OnMediaPlayerTimeChange;
+				_mediaPlayer.TimeChanged += OnMediaPlayerTimeChangeIsMediaParse;
+				_mediaPlayer.MediaChanged += MediaPlayerMediaChanged;
+				_mediaPlayer.Stopped += OnMediaPlayerStopped;
+
+				_videoContainer.Content = _videoView;
+
+				Child = _videoContainer;
+
+				UpdateVideoStretch();
+				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				{
+					this.Log().Debug($"Created player");
+				}
+				_taskCompletionSource.TrySetResult("");
+			}
+		}
+		catch (Exception ex)
+		{
+			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			{
+				this.Log().Debug($"InitializedMedia error: " + ex.Message);
+			}
```

Comment:
```suggestion
			}
			_taskCompletionSource.TrySetException(ex);
```

And you should also add support for `CancellationToken` and add a catch
```cs
catch (OperationCancelledException) when (ct.IsCancellationRequested)
{
    _taskCompletionSource.TrySetCancelled();
}
```

Reply:
Won't be need.
We remove the TaskCompletionSource